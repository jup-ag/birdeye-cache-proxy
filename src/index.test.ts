/// <reference path="../node_modules/@types/bun/index.d.ts" />

import { expect, test, describe, it } from 'bun:test';

describe('proxy birdeye requests', () => {
  it('should work', async () => {
    type PriceResponse = {
      value: number;
      updateUnixTime: number;
      updateHumanTime: string;
      priceChange24h: number;
    };
    const response = await fetch(
      'http://localhost:8787/defi/multi_price?list_address=So11111111111111111111111111111111111111112,27G8MtK7VtTcCHkpASjSDdkWWYfoqT6ggEuKidVJidD4',
    );
    const { data } = await response.json<{
      data: Record<string, PriceResponse>;
      success: true;
    }>();

    expect(data).toMatchSnapshot({
      So11111111111111111111111111111111111111112: {
        value: expect.any(Number),
        updateUnixTime: expect.any(Number),
        updateHumanTime: expect.any(String),
        priceChange24h: expect.any(Number),
      },
      '27G8MtK7VtTcCHkpASjSDdkWWYfoqT6ggEuKidVJidD4': {
        value: expect.any(Number),
        updateUnixTime: expect.any(Number),
        updateHumanTime: expect.any(String),
        priceChange24h: expect.any(Number),
      },
    });
  });
});
