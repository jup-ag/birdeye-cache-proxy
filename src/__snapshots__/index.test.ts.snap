// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`proxy birdeye requests should work 1`] = `
{
  "27G8MtK7VtTcCHkpASjSDdkWWYfoqT6ggEuKidVJidD4": {
    "priceChange24h": Any<Number>,
    "updateHumanTime": Any<String>,
    "updateUnixTime": Any<Number>,
    "value": Any<Number>,
  },
  "So11111111111111111111111111111111111111112": {
    "priceChange24h": Any<Number>,
    "updateHumanTime": Any<String>,
    "updateUnixTime": Any<Number>,
    "value": Any<Number>,
  },
}
`;
